### ISSUE TITLE
bad schema when using typing.Tuple

### ISSUE THREAD

**Issue by MikailBag** [tiangolo/fastapi#1507]
### Describe the bug

Write here a clear and concise description of what the bug is.

### To Reproduce

Steps to reproduce the behavior with a minimum self-contained file.

Replace each part with your own scenario:

1. Create a file with:

```python
from fastapi import FastAPI
import typing
import pydantic
import json

class Foo(pydantic.BaseModel):
    field: typing.Tuple[str, str]

app = FastAPI()


@app.get("/", response_model=Foo)
def read_root():
    # body does not matter
    pass

print(json.dumps(Foo.schema()))
```
3. Get schema from /openapi.json endpoint
4. Use the schema.
5. This schema will contain following definition for "components/Foo":
```json
{
    "title": "Foo",
    "type": "object",
    "properties": {
        "field": {
            "title": "Field",
            "type": "array",
            "items": [
                {
                    "type": "string"
                },
                {
                    "type": "string"
                }
            ]
        }
    },
    "required": [
        "field"
    ]
}
```
(as far as I understand, fastapi takes schema directly from pydantic).
6. This definition does not comply to OpenAPI spec:
```
items - Value MUST be an object and not an array. Inline or referenced schema MUST be of a Schema Object and not a standard JSON Schema. items MUST be present if the type is array.
```
[(source)](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#schemaObject)

### Expected behavior

I expect to get schema, conforming to OpenAPI specification, so that other tooling can successfully consume this schema.

### Environment

- OS: Ubuntu Linux 20.04
- FastAPI Version 0.54.2
- Python version 3.8.2

## Additional context
I am not sure if the bug should be really fixed on `fastapi` side.
Also reported (issue was closed as out-of-scope) on pydantic: samuelcolvin/pydantic/issues/1594

**Comment by tiangolo** [comment]
Yep, exactly, you cannot use `Tuple` with OpenAPI, you can only use `List` with a single sub-type.

The next version of OpenAPI, 3.1.0, will be based on the latest JSON Schema and will support it.

**Comment by saabeilin** [comment]
@tiangolo Any timeline when it will be supported in FastAPI?

**Comment by vdwees** [comment]
@saabeilin Tuples were actually supported in 0.67.0, but they broke in 0.68.0 (see #3665)
